tmc-langs-util/src/test/java/fi/helsinki/cs/tmc/langs/util/MainTest.java

```diff
@@ -0,0 +1,203 @@
+
+    @Rule
+    public MockStdio mio = new MockStdio();
+
+    @Rule
+    public final ExpectedSystemExit exit = ExpectedSystemExit.none();
+
+    private final String HELP_TEXT = "Usage: Main <exercise path> <output path>\n"
+        + "\nOptions:\n"
+        + " --checkstyle <exercise path> <output path>\t\tRun checkstyle or similar plugin to project if applicable.\n"
+        + " --help\t\t\t\t\t\t\tDisplay help information.\n"
+        + " --preparesolution <exercise path>\t\t\tPrepare a presentable solution from the original.\n"
+        + " --preparestub <exercise path>\t\t\t\tPrepare a stub exercise from the original.\n"
+        + " --runtests <exercise path> <output path>\t\tRun the tests for the exercise.\n"
+        + " --scanexercise <exercise path> <output path>\t\tProduce an exercise description of an exercise directory.";
+    private String expectedMessage = "";
+
+    @Test
+    public void testMain() {
+        String[] args = null;
+        exit.expectSystemExitWithStatus(0);
+        exitStringEqualsAssertion(HELP_TEXT);
+        Main.main(args);
+    }
+
+    @Test
+    public void testWithNoArgs() {
+        String[] args = {};
+        expectedMessage = HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testHelp() {
+        String[] args = {"--help"};
+        expectedMessage = HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testHelpWithInvalidArgumentCountOneArg() {
+        String[] args = {"--help", "dummy_string"};
+        expectedMessage = "ERROR: wrong argument count for --help expected 0 got 1\n\n" + HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testScanExercise() {
+        String exercisePath = getTargetPath("arith_funcs");
+        String outputPath = exercisePath + "/checkstyle.txt";
+        String[] args = {"--scanexercise", exercisePath, outputPath};
+        expectedMessage = "Exercises scanned successfully, results can be found in " + outputPath;
+        mainTest(0, args);
+    }
+
+    @Test
+    public void testScanExerciseWithInvalidArgs() {
+        String[] args = {"--scanexercise", "dummy string", "another"};
+        exit.expectSystemExitWithStatus(0);
+        expectedMessage = "ERROR: Given test path is not a directory.\n\n" + HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testScanExerciseWithInvalidArgumentCountZeroArgs() {
+        String[] args = {"--scanexercise"};
+        expectedMessage = "ERROR: wrong argument count for --scanexercise expected 2 got 0\n\n" + HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testScanExerciseWithInvalidArgumentCountOneArg() {
+        String[] args = {"--scanexercise", "dummy string"};
+        expectedMessage = "ERROR: wrong argument count for --scanexercise expected 2 got 1\n\n" + HELP_TEXT;
+        mainTestEquals(0, args);
+    }
+
+    @Test
+    public void testRunTests() {
```

Comment:
I think you should mock the executor in all tests. The functionality should be tested elsewhere, right?